### PR TITLE
load test matrix from json file

### PR DESCRIPTION
**Pull request by pdabelf5**
### Proposed changes

Refactor test setup matrix

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by oseoin — APPROVED** [review]

File: .github/data/matrix-regression.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "k8s": [
+    "1.22.17",
```
Comment (vepatel) [inline .github/data/matrix-regression.json:3]: this needs to be updated from @j1m-ryan's recent changes

Comment (j1m-ryan) [reply .github/data/matrix-regression.json:3]: I don't mind if we have the versions in code declaritvely like this in json. One advantage of this way is that we can probably automate that part of the release where we say what versions we tested it on, a little easier that if we are getting the versions from an API. But yeah we probably ought to add in the latest two verions here, 1.28.0 and 1.29.0

**Review by j1m-ryan — APPROVED** [review]

File: .github/data/matrix-regression.json (hunk 1 above)
Comment (oseoin) [reply .github/data/matrix-regression.json]: @vepatel @j1m-ryan that should pull the versions dynamically now, from a local run the generated matrix was:

```json
{
  "k8s": [
    [
      "1.23.17",
      "1.24.15",
      "1.25.11",
      "1.26.6",
      "1.27.3",
      "1.28.0",
      "1.29.0"
    ]
  ],
  "images": [
    {
      "image": "debian"
    },
    {
      "image": "debian-plus"
    }
  ]
}
```

Comment (oseoin) [reply .github/data/matrix-regression.json]: Working matrix from manual regression trigger with @pdabelf5:
image

**Review by vepatel — APPROVED** [review]

**Pull request merged**